### PR TITLE
snap: add `snap watch <change-id>` to attach to a running change

### PR DESCRIPTION
**Pull request by mvo5**
LP: #1634392

### CODE REVIEW

**Comment by mvo5** [comment]
@chipaca None at all, actually `watch` seems to be more appropriate. So unless @niemeyer disagrees I will rename it to `snap watch <change-id>` :)

File: cmd/snap/cmd_snap_op.go [hunk 1]
```diff
@@ -57,9 +57,15 @@ func wait(client *client.Client, id string) (*client.Change, error) {
 	var lastID string
 	lastLog := map[string]string{}
 	for {
-		chg, err := client.Change(id)
+		chg, err := cli.Change(id)
+		// a client.Error means we were able to communicate with
+		// the server (got an answer)
+		if e, ok := err.(*client.Error); ok {
```
Comment (zyga) [inline cmd/snap/cmd_snap_op.go]: Should this also test if `err != nil`?

File: cmd/snap/cmd_watch.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"github.com/snapcore/snapd/i18n"
```
Comment (zyga) [inline cmd/snap/cmd_watch.go]: Isn't the import other the other way around? (stdlib, 3rd-party, internal)

File: cmd/snap/cmd_snap_op.go [hunk 3]
```diff
@@ -46,7 +46,7 @@ var (
 	pollTime    = 100 * time.Millisecond
 )
 
-func wait(client *client.Client, id string) (*client.Change, error) {
+func wait(cli *client.Client, id string) (*client.Change, error) {
```
Comment (zyga) [inline cmd/snap/cmd_snap_op.go]: `id` is probably change ID but this is not apparent. Perhaps it could be a dedicated type?

File: cmd/snap/cmd_watch_test.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	. "gopkg.in/check.v1"
+
+	snap "github.com/snapcore/snapd/cmd/snap"
```
Comment (zyga) [inline cmd/snap/cmd_watch_test.go]: Isn't this a snap already?

File: cmd/snap/cmd_watch.go (hunk 2 above)
Comment (mvo5) [inline cmd/snap/cmd_watch.go]: Indeed, thank you!

File: cmd/snap/cmd_watch_test.go (hunk 4 above)
Comment (mvo5) [inline cmd/snap/cmd_watch_test.go]: This is needed because the package is "main" (the directory name is snap but all go files in there have `package main`)

File: cmd/snap/cmd_snap_op.go (hunk 3 above)
Comment (mvo5) [inline cmd/snap/cmd_snap_op.go]: Good point, I will leave that for a followup branch though
